tests/training/test_classification.py

```diff
@@ -431,6 +431,7 @@ def test_lgbm_classification_learner():
                                                  learning_rate=0.1,
                                                  num_estimators=20,
                                                  extra_params={"max_depth": 4, "seed": 42},
+                                                 categorical_features=["x2"],
```

Comment:
We should probably test that `x2` is actually being treated as categorical. We can check the split types of the booster object, because for categorical feature the decision type is '==' instead of '<='

Reply:
Thank you for the feedback! 

I added a new test case and a check similar to this:
`assert any(tree['num_cat'] > 0 for tree in log['object'].dump_model()['tree_info'])`

Please let me know if any other changes are necessary.